### DISCUSSION TITLE
Getter type

### DISCUSSION THREAD

**Discussion by wasya243** [typegoose/typegoose#855] [Q&A]
Hello, please take a look at my issue. Here is sample code.

```ts
const transformMapToObject = (purses: Map<string, UserPurse>): { [key: string]: UserPurse } => {
  const keys = Array.from(purses.keys());
  const result: { [key: string]: UserPurse } = {};

  for(const key of keys) {
    const purse = purses.get(key);

    if (purse) {
      result[key] = purse;
    }
  }

  return result;
}

// TS will think that type of this field is Map<string, UserPurse> when I reference it in code
// but in reality return type of this field is { [key: string]: UserPurse }, 'cause of getter

class User {
	@prop({ type: () => Map<string, UserPurse>, get: transforMapToObject })
    purses?: Map<string, UserPurse>;	
}

```

So, for legacy code reasons I assign Map<string, UserPurse> type to .purses field, but I need to get `{ [key: string]: UserPurse }` type when I reference .purses field. However, despite getter type TS still thinks it's Map<string, UserPurse> when I reference it. How do I fix it?

**Comment by hasezoey** [comment]
>  However, despite getter type TS still thinks it's Map<string, UserPurse> when I reference it. How do I fix it?

decorators cannot affect the typescript type, you either need to change the type from `Map<string, UserPurse>` directly or need to cast it somewhere where its needed (the most correct way in this case would be to change the current Map type)
